packages/cds-odata-annotation-converter/src/printer/csdl-to-cds.ts

```diff
@@ -128,9 +128,23 @@ export const printTarget = (target: Target): string => {
             //        <term> <qualifier>: <value>
             //     );
             // }
-            result = `annotate ${rootElementName} with {\n${childSegments[0]} ${
-                target.terms.length > 1 ? `@(${result})` : `@${result}`
-            }};\n`;
+            const terms = target.terms.length > 1 ? `@(${result})` : `@${result}`;
+            if (isStructureTargetType) {
+                let closingBraces: string[] = [];
+                const complexTarget = childSegments[0].split('_').reduce((acc, item, index, array) => {
```

Comment:
We should not try to rebuild the structure here, because we don't have all the information here. Instead the structure should be passed as a parameter.

Reply:
adapted